src/otx/engine/engine.py

```diff
@@ -572,29 +572,47 @@ def explain(
         """
         from otx.algo.utils.xai_utils import process_saliency_maps_in_pred_entity
 
-        ckpt_path = str(checkpoint) if checkpoint is not None else self.checkpoint
-        if explain_config is None:
-            explain_config = ExplainConfig()
+        model = self.model
+
+        if checkpoint is not None:
+            checkpoint = str(checkpoint)
+        elif self.checkpoint is not None:
+            checkpoint = str(self.checkpoint)
+        else:
+            checkpoint = None
+
+        datamodule = datamodule if datamodule is not None else self.datamodule
+
+        is_ir_ckpt = checkpoint is not None and Path(checkpoint).suffix in [".xml", ".onnx"]
+        if is_ir_ckpt and not isinstance(model, OVModel):
+            datamodule = self._auto_configurator.get_ov_datamodule()
+            model = self._auto_configurator.get_ov_model(model_name=str(checkpoint), label_info=datamodule.label_info)
 
         lit_module = self._build_lightning_module(
-            model=self.model,
+            model=model,
             optimizer=self.optimizer,
             scheduler=self.scheduler,
         )
-        if datamodule is None:
-            datamodule = self.datamodule
         lit_module.label_info = datamodule.label_info
 
+        # NOTE, trainer.test takes only lightning based checkpoint.
+        # So, it can't take the OTX1.x checkpoint.
```

Comment:
As i know, we can load 1.X checkpoint with this code.
@sungmanc Is it right?

Reply:
Yes, we could load the OTX1.X checkpoint.

Reply:
Thank you, removed this comment